tests/test_batch.py

```diff
@@ -68,6 +76,46 @@ def test_create_batch_and_fetch_results(self, request_mock):
             assert job.result == self.job_result
         assert request_mock.last_request.method == "GET"
 
+    def test_get_batch(self, request_mock):
+        batch_requested = self.sdk.get_batch(self.batch.id)
+        assert (
+            batch_requested.id == self.batch.id
+        )
+
+    def test_cancel_batch_endpoint(self, request_mock):
+        self.batch.cancel()
+        assert request_mock.last_request.method == "PUT"
+        assert (
+            request_mock.last_request.url
+            == f"{self.sdk._client.endpoints.core}/api/v1/batches/{self.batch.id}/cancel"
+        )
+
+    def test_cancel_batch_success(self, request_mock):
+        self.batch.cancel()
+        assert (
+            self.batch.status == 'CANCELED'
+        )
+
+    def test_add_job_and_get_job(self, request_mock):
+        job = self.batch.add_job(
+            runs=self.n_job_runs,
+            variables=self.job_variables,
+        )
+        # the name of the dictionary is the id of the job
+        job_just_added_id = list(self.batch.jobs.keys())[-1]
+        job_requested = self.sdk.get_job(job.id)
+        assert (
+            job_requested.id == job_just_added_id
+        )
+
+    def test_cancel_job_endpoint(self, request_mock):
+        self.batch.cancel_job(self.job_id)
```

Comment:
- Assert the job status
- Don't forget to test both the SDK class cancel_job method and the Job class cancel method (after you add it)